src/v/s3/client.cc

```diff
@@ -181,6 +185,32 @@ result<http::client::request_header> request_creator::make_get_object_request(
     return header;
 }
 
+result<http::client::request_header> request_creator::make_head_object_request(
+  bucket_name const& name, object_key const& key) {
+    http::client::request_header header{};
+    // HEAD /{object-id} HTTP/1.1
+    // Host: {bucket-name}.s3.amazonaws.com
+    // x-amz-date:{req-datetime}
+    // Authorization:{signature}
+    // x-amz-content-sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
+    auto host = fmt::format("{}.{}", name(), _ap());
+    auto target = fmt::format("/{}", key().string());
+    std::string emptysig
+      = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855";
```

Comment:
where did we get this string from?

Reply:
make it const?

Reply:
also confused (and agree with constexpr). I'm assuming this is part of the AWS API, and I see this same sha256 repeated above in make_get_object_request so maybe worth factoring out with a comment / reference into documentation?